packages/mds-provider-processor/tests/metrics.test.ts

```diff
@@ -0,0 +1,82 @@
+import db from '@mds-core/mds-db'
+import cache from '@mds-core/mds-cache'
+import { now } from '@mds-core/mds-utils'
+import { VEHICLE_EVENTS, VEHICLE_EVENT, VEHICLE_TYPES } from '@mds-core/mds-types'
+import uuid from 'uuid'
+import assert from 'assert'
+import metric from '../src/metrics'
+import config from '../src/config'
+
+const PROVIDER_ID = config.organization.providers[0]
+const time = now()
+async function seedDeviceState(event: VEHICLE_EVENT) {
+  await db.insertDeviceStates({
+    vehicle_type: VEHICLE_TYPES.scooter,
+    type: 'event',
+    timestamp: time,
+    device_id: uuid(),
+    provider_id: PROVIDER_ID,
+    recorded: now(),
+    annotation_version: 1,
+    annotation: null,
+    gps: null,
+    service_area_id: null,
+    charge: null,
+    state: null,
+    event_type: event,
+    event_type_reason: null,
+    trip_id: uuid()
+  })
+}
+async function seedDeviceStates() {
+  await Promise.all(
+    Object.keys(VEHICLE_EVENTS).map(async event => {
+      await seedDeviceState(event as VEHICLE_EVENT)
+    })
+  )
+  await seedDeviceState(VEHICLE_EVENTS.register)
+  await seedDeviceState(VEHICLE_EVENTS.register)
+}
+
+describe('Proc Metrics', () => {
+  before(async () => {
+    await db.initialize()
+    await seedDeviceStates()
+  })
+
+  after(async () => {
+    await db.shutdown()
+    await cache.shutdown()
+  })
+
+  it('calcEventCounts', async () => {
```

Comment:
This `async` unit test pattern is so much cleaner. We should stop writing tests using a `done()` callback.